### ISSUE TITLE
Crashes in PermissionsModule.onRequestPermissionsResult 

### ISSUE THREAD

**Issue by thehellmaker** [zoontek/react-native-permissions#820]
### Bug summary

Referencing a similar issue reported on react-native  https://github.com/facebook/react-native/issues/29056
The issue was closed stating unable to reproduce. I believe this issue is not easily reproduceable one since it doesn't happen always. We are seeing 3% crashes because of this issue. From the logs it seems like is happens when the app is resumed from background. onResume is causing this issue. i think it would be great to have some defensive check on 

```java
Fatal Exception: java.lang.RuntimeException
Unable to resume activity {com.example.app/com.example.app.MainActivity}: java.lang.NullPointerException: Attempt to invoke interface method 'void com.facebook.react.bridge.Callback.invoke(java.lang.Object[])' on a null object reference
Fatal Exception: java.lang.RuntimeException: Unable to resume activity {com.example.app/com.example.app.MainActivity}: java.lang.NullPointerException: Attempt to invoke interface method 'void com.facebook.react.bridge.Callback.invoke(java.lang.Object[])' on a null object reference
       at android.app.ActivityThread.performResumeActivity(ActivityThread.java:5070)
       at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:5113)
       at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:60)
       at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45)
       at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:176)
       at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2470)
       at android.os.Handler.dispatchMessage(Handler.java:106)
       at android.os.Looper.loopOnce(Looper.java:240)
       at android.os.Looper.loop(Looper.java:351)
       at android.app.ActivityThread.main(ActivityThread.java:8423)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:584)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1013)

Caused by java.lang.NullPointerException: Attempt to invoke interface method 'void com.facebook.react.bridge.Callback.invoke(java.lang.Object[])' on a null object reference
       at com.zoontek.rnpermissions.RNPermissionsModule.onRequestPermissionsResult(RNPermissionsModule.java:416)
       at com.facebook.react.ReactActivityDelegate$2.invoke(ReactActivityDelegate.java:198)
       at com.facebook.react.ReactActivityDelegate.onResume(ReactActivityDelegate.java:123)
       at com.facebook.react.ReactActivity.onResume(ReactActivity.java:58)
       at android.app.Instrumentation.callActivityOnResume(Instrumentation.java:1550)
       at android.app.Activity.performResume(Activity.java:8565)
       at android.app.ActivityThread.performResumeActivity(ActivityThread.java:5052)
       at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:5113)
       at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:60)
       at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45)
       at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:176)
       at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2470)
       at android.os.Handler.dispatchMessage(Handler.java:106)
       at android.os.Looper.loopOnce(Looper.java:240)
       at android.os.Looper.loop(Looper.java:351)
       at android.app.ActivityThread.main(ActivityThread.java:8423)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:584)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1013)
```

### Library version

3.10.0

### Environment info

```shell
npx react-native info
info Fetching system and libraries information...
System:
    OS: macOS 14.0
    CPU: (12) x64 Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
    Memory: 64.69 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.8.0 - ~/.nvm/versions/node/v20.8.0/bin/node
    Yarn: 1.22.1 - /usr/local/bin/yarn
    npm: 10.1.0 - ~/.nvm/versions/node/v20.8.0/bin/npm
    Watchman: 2023.10.02.00 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.13.0 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 23.0, iOS 17.0, macOS 14.0, tvOS 17.0, watchOS 10.0
    Android SDK: Not Found
  IDEs:
    Android Studio: 2022.3 AI-223.8836.35.2231.10811636
    Xcode: 15.0/15A240d - /usr/bin/xcodebuild
  Languages:
    Java: 11.0.20.1 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: Not Found
    react-native: Not Found
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
```

### Steps to reproduce

No clear steps to reproduce. It happens erratically for small percent of users. can we add some null check to handle this error gracefully and let the caller handle this on JS side rather than an App crash which the developer had no control over? I would be happy to open a PR for the same.

### Reproducible sample code

```js
No clear steps to reproduce. It happens erratically for small percent of users. can we add some null check to handle this error gracefully and let the caller handle this on JS side rather than an App crash which the developer had no control over? I would be happy to open a PR for the same.
```

**Comment by zoontek** [comment]
Do you have analytics / infos about the crashing devices? Android version / manufacturers (sometimes highly modified fork like MIUI, suffers from their own issues).
Because it it's not reproductible, I'm not sure we can do more than the Facebook team (the android native part of this module is roughly the same as `PermissionsAndroid`)

**Comment by zoontek** [comment]
Should do the trick: https://github.com/zoontek/react-native-permissions/releases/tag/3.10.1

**Comment by thehellmaker** [comment]
<img width="536" alt="Screenshot 2023-10-26 at 2 50 36 PM" src="https://github.com/zoontek/react-native-permissions/assets/1051911/158f94a5-e25b-408f-8093-348d852557f3">

**Comment by zoontek** [comment]
Yeah, so two brands only + 100% Android 13. Probably a bad implem on their side. The new `catch` will avoid crashes when this occur.

**Comment by thehellmaker** [comment]
Thanks a lot for the 3.10.1 quick fix. That should work. Would you also be able to push this change to npm?

**Comment by zoontek** [comment]
@thehellmaker Already done: https://www.npmjs.com/package/react-native-permissions?activeTab=versions

**Comment by podger** [comment]
Same problem here with little occurence rate but only on Samsung so far in production. Also once occured on one Pixel 6 during beta-testing (just once) and couldn't reproduce anymore.

I dived already into 3.10.1 with the trick of catching the NullException but...

Isn't this masking another issue that will result in a silent denied permission? The experienced we had during testing was that a crash happened **before any prompt was asked to the user**. If the 3.10.1 fix would occur then I think that would just continue with all permissions denied. Am I wrong?

**Comment by zoontek** [comment]
@podger Probably, but I'm not sure why the OS crashed like this, as I don't have a solid reproduction it's the best we can do for the moment.

**Comment by nppull** [comment]
Crash on samsung galaxy A22 - Android 11, API 30
React native 0.72.7
React native permission 4.0.0

-- my solution --
```
import {
  PermissionsAndroid
} from 'react-native';
```

**Comment by aguglie** [comment]
Hello 👋,

I’m currently troubleshooting an issue with `react-native-permissions:4.1.5`. From the crashlytics log it happened on a OnePlus device running Android 14. This problem was previously marked as resolved in version 3.0.1, but it appears to persist. I noticed that the catch block added as a fix in [this commit](https://github.com/zoontek/react-native-permissions/blob/04bf499c7cd4ac4b6a43f4ae62cc8719f905abf5/android/src/main/java/com/zoontek/rnpermissions/RNPermissionsModuleImpl.java#L358) is still intact, yet a `NullPointerException` is still occurring.

I am attaching the stack trace for more context. Could you provide any hints or point me towards potential areas I might investigate to better understand the root cause of this issue? Any insights would be very helpful!

```
Caused by java.lang.NullPointerException: Attempt to invoke interface method 'void com.facebook.react.bridge.Callback.invoke(java.lang.Object[])' on a null object reference
       at com.facebook.react.modules.permissions.PermissionsModule.onRequestPermissionsResult(PermissionsModule.java:201)
       at com.facebook.react.ReactActivityDelegate$2.invoke(ReactActivityDelegate.java:201)
       at com.facebook.react.ReactActivityDelegate.onResume(ReactActivityDelegate.java:126)
       at java.lang.reflect.Method.invoke(Method.java)
       at expo.modules.ReactActivityDelegateWrapper.invokeDelegateMethod(ReactActivityDelegateWrapper.kt:285)
       at expo.modules.ReactActivityDelegateWrapper.onResume(ReactActivityDelegateWrapper.kt:163)
       at com.facebook.react.ReactActivity.onResume(ReactActivity.java:58)
       at android.app.Instrumentation.callActivityOnResume(Instrumentation.java:1622)
       at android.app.Activity.performResume(Activity.java:8916)
       at android.app.ActivityThread.performResumeActivity(ActivityThread.java:5264)
       at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:5325)
       at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:64)
       at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45)
       at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:185)
       at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:103)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2630)
       at android.os.Handler.dispatchMessage(Handler.java:106)
       at android.os.Looper.loopOnce(Looper.java:257)
       at android.os.Looper.loop(Looper.java:368)
       at android.app.ActivityThread.main(ActivityThread.java:8826)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:572)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1049)
```

**Comment by adriangonzy** [comment]
Yep I can confirm the same exact issue but I'm on a expo app. Happens just after accepting the permissions the first time and then when I relaunch the app it works fine. 
The phone is a One Plus 9 with Android 14 too.

java.lang.RuntimeException: Unable to resume activity {blabla/blabla.MainActivity}: java.lang.reflect.InvocationTargetException
    at android.app.ActivityThread.performResumeActivity(ActivityThread.java:5282)
    at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:5325)
    at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:64)
    at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45)
    at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:185)
    at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:103)
    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2630)
    at android.os.Handler.dispatchMessage(Handler.java:106)
    at android.os.Looper.loopOnce(Looper.java:257)
    at android.os.Looper.loop(Looper.java:368)
    at android.app.ActivityThread.main(ActivityThread.java:8826)
    at java.lang.reflect.Method.invoke(Method.java)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:572)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1049)
java.lang.reflect.InvocationTargetException: null
    at java.lang.reflect.Method.invoke(Method.java)
    at expo.modules.ReactActivityDelegateWrapper.invokeDelegateMethod(ReactActivityDelegateWrapper.kt:294)
    at expo.modules.ReactActivityDelegateWrapper.onResume(ReactActivityDelegateWrapper.kt:168)
    at com.facebook.react.ReactActivity.onResume(ReactActivity.java:58)
    at android.app.Instrumentation.callActivityOnResume(Instrumentation.java:1622)
    at android.app.Activity.performResume(Activity.java:8916)
    at android.app.ActivityThread.performResumeActivity(ActivityThread.java:5264)
    at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:5325)
    at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:64)
    at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45)
    at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:185)
    at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:103)
    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2630)
    at android.os.Handler.dispatchMessage(Handler.java:106)
    at android.os.Looper.loopOnce(Looper.java:257)
    at android.os.Looper.loop(Looper.java:368)
    at android.app.ActivityThread.main(ActivityThread.java:8826)
    at java.lang.reflect.Method.invoke(Method.java)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:572)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1049)
java.lang.NullPointerException: Attempt to invoke interface method 'void com.facebook.react.bridge.Callback.invoke(java.lang.Object[])' on a null object reference
    at com.facebook.react.modules.permissions.PermissionsModule.onRequestPermissionsResult(PermissionsModule.java:203)
    at com.facebook.react.ReactActivityDelegate$2.invoke(ReactActivityDelegate.java:220)
    at com.facebook.react.ReactActivityDelegate.onResume(ReactActivityDelegate.java:135)
    at java.lang.reflect.Method.invoke(Method.java)
    at expo.modules.ReactActivityDelegateWrapper.invokeDelegateMethod(ReactActivityDelegateWrapper.kt:294)
    at expo.modules.ReactActivityDelegateWrapper.onResume(ReactActivityDelegateWrapper.kt:168)
    at com.facebook.react.ReactActivity.onResume(ReactActivity.java:58)
    at android.app.Instrumentation.callActivityOnResume(Instrumentation.java:1622)
    at android.app.Activity.performResume(Activity.java:8916)
    at android.app.ActivityThread.performResumeActivity(ActivityThread.java:5264)
    at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:5325)
    at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:64)
    at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45)
    at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:185)
    at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:103)
    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2630)
    at android.os.Handler.dispatchMessage(Handler.java:106)
    at android.os.Looper.loopOnce(Looper.java:257)
    at android.os.Looper.loop(Looper.java:368)
    at android.app.ActivityThread.main(ActivityThread.java:8826)
    at java.lang.reflect.Method.invoke(Method.java)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:572)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1049)

**Comment by aguglie** [comment]
@adriangonzy I dug further into the issue and found that in my case, it was not caused by `react-native-permissions` but rather originated from the native `PermissionsAndroid`. As Zoontek mentioned, the `PermissionsAndroid` module functions similarly to this one. They addressed the issue with the same fix, which can be seen [here](https://github.com/facebook/react-native/commit/925209932aa20f6c70e9642c5ab8dec8ac83a1cd). Therefore, updating to `react-native v0.74.0` or patching the `PermissionsModule.java` as outlined in the commit should resolve the problem.

Edit: Errata fixed on v0.74.0

**Comment by GSSPawanKumarSingh** [comment]
Can anyone please share the patch which need to done for version 0.73.8 with steps.?

> @adriangonzy I dug further into the issue and found that in my case, it was not caused by `react-native-permissions` but rather originated from the native `PermissionsAndroid`. As Zoontek mentioned, the `PermissionsAndroid` module functions similarly to this one. They addressed the issue with the same fix, which can be seen [here](https://github.com/facebook/react-native/commit/925209932aa20f6c70e9642c5ab8dec8ac83a1cd). Therefore, updating to `react-native v0.74.0` or patching the `PermissionsModule.java` as outlined in the commit should resolve the problem.
> 
> Edit: Errata fixed on v0.74.0

**Comment by muhammad-kumail** [comment]
> [@adriangonzy](https://github.com/adriangonzy) I dug further into the issue and found that in my case, it was not caused by `react-native-permissions` but rather originated from the native `PermissionsAndroid`. As Zoontek mentioned, the `PermissionsAndroid` module functions similarly to this one. They addressed the issue with the same fix, which can be seen [here](https://github.com/facebook/react-native/commit/925209932aa20f6c70e9642c5ab8dec8ac83a1cd). Therefore, updating to `react-native v0.74.0` or patching the `PermissionsModule.java` as outlined in the commit should resolve the problem.
> 
> Edit: Errata fixed on v0.74.0

I'm using v0.76.2 which has already has same fix-code as use mentioned. But still app crashing whenever permission request app crashes.

**Comment by amitkumarcoding** [comment]
> > [@adriangonzy](https://github.com/adriangonzy) I dug further into the issue and found that in my case, it was not caused by `react-native-permissions` but rather originated from the native `PermissionsAndroid`. As Zoontek mentioned, the `PermissionsAndroid` module functions similarly to this one. They addressed the issue with the same fix, which can be seen [here](https://github.com/facebook/react-native/commit/925209932aa20f6c70e9642c5ab8dec8ac83a1cd). Therefore, updating to `react-native v0.74.0` or patching the `PermissionsModule.java` as outlined in the commit should resolve the problem.
> > Edit: Errata fixed on v0.74.0
> 
> I'm using v0.76.2 which has already has same fix-code as use mentioned. But still app crashing whenever permission request app crashes.


same here

**Comment by gabk17** [comment]
@muhammad-kumail @amitkumarcoding Did u figure this one out?
